doc/read-the-docs-site/extensions-flags-pragmas.rst

```diff
@@ -0,0 +1,83 @@
+Although Plutus Tx is, syntactically, a subset of Haskell, it has different semantics and a different evaluation strategy (Haskell: non-strict semantics, call by need; Plutus Tx: strict semantics, call by value).
+As a result, some GHC optimizations are not helpful for Plutus Tx programs, and can even be harmful, in the sense that it can make Plutus Tx programs less efficient, or fail to be compiled.
+An example is the full laziness optimization, controlled by GHC flag ``-ffull-laziness``, which floats let bindings out of lambdas whenever possible.
+Since Untyped Plutus Core does not employ lazy evaluation, the full laziness optimization is never beneficial, and can sometimes make a Plutus Tx program more expensive.
+Conversely, some GHC features must be turned on in order to ensure Plutus Tx programs are compiled successfully.
+
+All Plutus Tx modules should use the following GHC flags: ::
+
+  -fplugin PlutusTx.Plugin
+  -fno-ignore-interface-pragmas
+  -fno-omit-interface-pragmas
+  -fno-full-laziness
+  -fno-spec-constr
+  -fno-specialise
+  -fno-strictness
+  -fno-unbox-strict-fields
+  -fno-unbox-small-strict-fields
+
+``-fplugin PlutusTx.Plugin`` enables the Plutus Tx compiler.
+``-fno-ignore-interface-pragmas`` and ``-fno-omit-interface-pragmas`` ensure unfoldings of Plutus Tx functions are available.
+The rest are GHC optimizations that are bad for Plutus Tx, and should thus be turned off.
+
+Note that this section only covers GHC flags, not Plutus Tx compiler flags.
+Information about the latter can be found in :ref:`plutus_tx_options`.
+
+Pragmas
+-------------------------------------------------
+
+All functions and methods should have the ``INLINEABLE`` pragma, so that their unfoldings are made available to the Plutus Tx compiler.
+
+The ``-fexpose-all-unfoldings`` flag also makes GHC expose all unfoldings, but unfoldings exposed this way can be more optimized than unfoldings exposed via ``INLINEABLE``.
+In general we do not want GHC to perform optimizations, since GHC optimizes a program based on the assumption that it has non-strict semantics and is evaluated lazily (call by need), which is not true for Plutus Tx programs.
+Therefore, ``INLINEABLE`` is preferred over ``-fexpose-all-unfoldings`` even though the latter is simpler.
+
+``-fexpose-all-unfoldings`` can be useful for functions that are generated by GHC and do not have the ``INLINEABLE`` pragma.
+``-fspecialise`` and ``-fspec-constr`` are two examples of optimizations that can generate such functions.
+The most reliable solution, however, is to simply turn these optimizations off.
+Another option is to bump ``-funfolding-creation-threshold`` to make it more likely for GHC to retain unfoldings for functions without the ``INLINEABLE`` pragma.
+``-fexpose-all-unfoldings`` should be used as a last resort.
+
+There's also the ``INLINE`` pragma, which does more than ``INLINEABLE`` in that the ``INLINE`` pragma forces the Plutus IR optimizer to inline the function.
```

Comment:
Does it? The Plutus IR one? Even on the Haskell side `INLINE` may not cause the definition to get inlined without an additional call to `inline` or at all (sometimes for a good reason like the definition being recursive).

Reply:
Yes, it indeed forces the Plutus IR inliner to inline the function: https://github.com/input-output-hk/plutus/pull/5183

Reply:
Impressive! Thanks for explaining.